### ISSUE TITLE
Fails on 32 bit systems (Date-Easy-0.01_01-TRIAL)

### ISSUE THREAD

**Issue by eserte** [barefootcoder/date-easy#3]
The following fail pattern seems to happen only on 32 bit systems: 
```
# ########################################
# TIMEZONE INFORMATION:
# 
#      Local Time: Sat May 14 17:02:05 2016
# Zone Specifiers: UTC +0000
#        Zonefile: UTC
#   Zonefile Info: timezone data, version 2, 8 gmt time flags, 8 std time flags, no leap seconds,
#                  144 transition times, 8 abbreviation chars
# ########################################
t/00-zoneinfo.t ........ skipped: Informational Only
t/date-math.t .......... ok

#   Failed test 'successful parse: 28991231'
#   at t/date-parse.t line 39.
#          got: '1901-12-13'
#     expected: '2899-12-31'

#   Failed test 'successful parse: 10000101'
#   at t/date-parse.t line 39.
#          got: '1901-12-13'
#     expected: '1000-01-01'
# Looks like you failed 2 tests of 733.
t/date-parse.t ......... 
Dubious, test returned 2 (wstat 512, 0x200)
Failed 2/733 subtests 

#   Failed test 'successfully constructed: 20520229'
#   at t/date.t line 34.
#          got: '19011213'
#     expected: '20520229'
# Looks like you failed 1 test of 17.
t/date.t ............... 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/17 subtests 
t/datetime-parse.t ..... ok

#   Failed test 'successfully constructed (6args): 20520229000000'
#   at t/datetime.t line 62.
#          got: '19011213204552'
#     expected: '20520229000000'
# Looks like you failed 1 test of 80.
t/datetime.t ........... 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/80 subtests 
```

**Comment by barefootcoder** [comment]
First of all, thanks very much for pinning this one down for me.  I hadn't really managed to pick this out of the recent raft of failures, because I think the majority of them are issue #4.  But this one is separate and I probably wouldn't have seen it until I got the others out of the way.

Secondly: Aw, crap. :-/  I had previously theorized that 32-bit systems might not have the same wide date range (i.e. from 1/1/1000 to 12/31/2899) as my test systems, but I didn't think about the fact that that meant my edge-case testing was going to fail on those systems.  Crap.

I'm not sure how to fix this, but at least now that I know what's going on I can work on it.  Thx again.

**Comment by barefootcoder** [comment]
Taking a crack at this one based on some basic Googling for how to detect a 32-bit machine.  Not 100% sure of the solution, though, so I'm going to wait for a verdict from CPAN Testers before closing the issue.

Also note that I really should have replaced my upper-and-lower-bound testing with appropriate upper and lower bounds for 32-bit machines.  But I'm not 100% sure what those are (although of course I have reasonable guesses).  I'm going to try to get access to a 32-bit machine myself so I can verify those and then maybe update those unit tests again.

**Comment by eserte** [comment]
The new TRIAL looks better. I managed to generate a pass report on my 32bit machine.

The only thing are these warnings which I don't see on 64bit systems:
```
Argument "1901-99-99" isn't numeric in numeric lt (<) at t/date-parse.t line 42.
Argument "1970-04-26" isn't numeric in numeric lt (<) at t/date-parse.t line 42.
Argument "2038-00-00" isn't numeric in numeric gt (>) at t/date-parse.t line 42.
Argument "2899-12-31" isn't numeric in numeric lt (<) at t/date-parse.t line 42.
Argument "2015-03-15" isn't numeric in numeric lt (<) at t/date-parse.t line 42.
Argument "1966-11-05" isn't numeric in numeric lt (<) at t/date-parse.t line 42.
Argument "1000-01-01" isn't numeric in numeric lt (<) at t/date-parse.t line 42.
Argument "1970-12-02" isn't numeric in numeric lt (<) at t/date-parse.t line 42.
Argument "2012-09-30" isn't numeric in numeric lt (<) at t/date-parse.t line 42.
t/date-parse.t ......... ok
```

**Comment by barefootcoder** [comment]
> The only thing are these warnings which I don't see on 64bit systems:

Yeah, I prolly should have used `le` and `ge` rather than `<` and `>`.  I'll change that and see if it fixes that warning.

Thx again for the extremely helpful testing.

**Comment by barefootcoder** [comment]
Should be fixed now.  Give it a try.

**Comment by barefootcoder** [comment]
I believe this is fixed in 0.02.  Closing out.
